### PR TITLE
Implement account access check and waiting room

### PR DESCRIPTION
**Pull request by kimlisa**
part of https://github.com/gravitational/teleport/issues/4384

#### Description
- Implement AccessRequest related components:
  - pending dialog
  - reason dialog
- Implement access request services (get/create access request)
- Tested the loader in mac (safari/google/firefox) and linux (google/firefox)
- Tested render cases

#### Related PR
- merge after all backend PRs of issue has been merged

#### Screenshot
All states available in storybook `Teleport > Access Request`

require reason
Screenshot from 2020-10-21 18-23-05

require reason error (when creating request)
Screenshot from 2020-10-21 18-23-12

awaiting for approval
Screenshot from 2020-10-21 18-22-47

error:
Screenshot from 2020-11-02 11-58-12

access denied with reason
Screenshot from 2020-10-21 18-22-41

access denied without reason (it's not required to provide a denial reason)
Screenshot from 2020-10-29 09-41-32

### CODE REVIEW

File: packages/teleport/src/components/AccessRequest/AccessRequest.tsx [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+    requestId,
+    request,
+    access,
+    getRequest,
+    createRequest,
+    renewSession,
+    removeUrlRequestParam,
+  } = props;
+
+  function render() {
```
Comment (alex-kovoy) [inline packages/teleport/src/components/AccessRequest/AccessRequest.tsx:54]: why this render() function?

File: packages/teleport/src/components/AccessRequest/AccessRequest.tsx [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+      return <AccessRequestPending getRequest={getRequest} />;
+    }
+
+    return <>{children}</>;
+  }
+
+  return render();
+}
+
+function AccessDenied({ reason, logout }: AccessDeniedProps) {
```
Comment (alex-kovoy) [inline packages/teleport/src/components/AccessRequest/AccessRequest.tsx:108]: Lets move < AccessDenied/> to a separate file

File: packages/teleport/src/components/AccessRequest/AccessRequestReason/AccessRequestReason.tsx [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        >
+          Send Request
+        </ButtonPrimary>
+        <ButtonSecondary onClick={logout}>Cancel & Logout</ButtonSecondary>
+      </DialogFooter>
+    </Dialog>
+  );
+}
+
+type Props = {
```
Comment (alex-kovoy) [inline packages/teleport/src/components/AccessRequest/AccessRequestReason/AccessRequestReason.tsx:94]: Lets move the `Props` to the hook and import it from there 

import useAccessRequestReason, { Props } from './useAccessRequestReason';

File: packages/teleport/src/components/AccessRequest/AccessRequest.tsx (hunk 1 above)
Comment (kimlisa) [reply packages/teleport/src/components/AccessRequest/AccessRequest.tsx]: removed (i am not sure why I did that in first place)

File: packages/teleport/src/components/AccessRequest/AccessRequest.tsx (hunk 2 above)
Comment (kimlisa) [reply packages/teleport/src/components/AccessRequest/AccessRequest.tsx]: separated

File: packages/teleport/src/components/AccessRequest/AccessRequestReason/AccessRequestReason.tsx (hunk 3 above)
Comment (kimlisa) [reply packages/teleport/src/components/AccessRequest/AccessRequestReason/AccessRequestReason.tsx]: moved container props to its hooks

**Review by alex-kovoy — APPROVED** [review]

**Pull request merged**